src/Utils/ProtoHelpers.php

```diff
@@ -39,4 +44,95 @@ public static function AddProto($desc, $proto)
         $desc->underlyingProto = $proto;
         return $desc;
     }
+
+    // Return type is dependant on option type. Either string, int, or Vector of string or int,
+    // or null if not repeated and value doesn't exist. Repeated returns empty vector if not exists.
+    private static function getCustomOptionRaw(Message $message, int $optionId, bool $repeated)
+    {
+        static $messageUnknown;
+        if (!$messageUnknown)
+        {
+            $ref = new \ReflectionClass('Google\Protobuf\Internal\Message');
+            $messageUnknown = $ref->getProperty('unknown');
+            $messageUnknown->setAccessible(true);
+        }
+
+        $values = [];
+        if ($message->hasOptions())
```

Comment:
Nit: Have we considered breaking out parts of this into other methods? Asking because the nesting looks a bit deep.

Reply:
Thanks for the thought - I think I'll leave it as-is for now, and will consider splitting it if it becomes too unwieldy later.